### ISSUE TITLE
Static (Unmanaged) VPC Placement has Messy Tag Handling

### ISSUE THREAD

**Issue by luthermonson** [kubernetes-sigs/cluster-api-provider-aws#3692]
/kind bug

**What steps did you take and what happened:**
Create two clusters with the same static VPC/Subnets/NAT Gateways
 - CAPA will reconcile both clusters and only the values of the most recently reconciled cluster will apply leading to them playing stomp out every reconciliation loop. This applies to AdditionalTags too which could get dicey if you think of something like a billing tag where you get them flipping the value back and forth.
 - When you delete these clusters there is code in delete VPC/Subnet/NAT Gateways to return nil immediately as CAPA didn't create them, this leaves the tags all in place and clutters up the VPC.

**What did you expect to happen:**
Since CAPA was told to not manage the VPC I don't think we should do the tag reconciliation if the user is trying to manage it themselves.

**Anything else you would like to add:**
I would like to add code to the reconciliation loops which looks like this code which will not reconcile tags if the VPC is unmanaged. Any feedback on that is appreciated.

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by Skarlso** [comment]
I agree that if the VPC is unmanaged, CAPA shouldn't do anything with it, including messing with the tags even if they are `AdditionalTags`.

That, however, should be clearly documented and is probably a "breaking change". If this feature has been relied upon previsouly.

**Comment by Skarlso** [comment]
Hmm, this is interesting. I can't really find the code where additional tags are applied to unmanaged VPCs.

@luthermonson can you give me the cluster yaml you used to replicate this problem?

**Comment by Skarlso** [comment]
One interesting point might be this... In order for the VPC to become unmanaged, it also needs to be missing the Owned tag:

```go
// IsUnmanaged returns true if the VPC is unmanaged.
func (v *VPCSpec) IsUnmanaged(clusterName string) bool {
	return v.ID != "" && !v.Tags.HasOwned(clusterName)
}

```

Maybe this tag was present in your VPC?
